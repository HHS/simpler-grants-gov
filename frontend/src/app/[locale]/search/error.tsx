"use client";

import QueryProvider from "src/app/[locale]/search/QueryProvider";
import { usePrevious } from "src/hooks/usePrevious";
import { FrontendErrorDetails } from "src/types/apiResponseTypes";
import { OptionalStringDict } from "src/types/generalTypes";
import { Breakpoints, ErrorProps } from "src/types/uiTypes";
import { convertSearchParamsToProperTypes } from "src/utils/search/convertSearchParamsToProperTypes";

import { useTranslations } from "next-intl";
import { ReadonlyURLSearchParams, useSearchParams } from "next/navigation";
import { useEffect } from "react";
import { Alert } from "@trussworks/react-uswds";

import ContentDisplayToggle from "src/components/ContentDisplayToggle";
import SearchBar from "src/components/search/SearchBar";
import SearchFilters from "src/components/search/SearchFilters";
import ServerErrorAlert from "src/components/ServerErrorAlert";

export interface ParsedError {
  message: string;
  searchInputs: OptionalStringDict;
  status: number;
  type: string;
  details?: FrontendErrorDetails;
}

function isValidJSON(str: string) {
  try {
    JSON.parse(str);
    return true;
  } catch (e) {
    return false; // String is not valid JSON
  }
}

function createBlankParsedError(): ParsedError {
  return {
    type: "NetworkError",
    searchInputs: {
      query: "",
      status: "",
      fundingInstrument: "",
      eligibility: "",
      agency: "",
      category: "",
      sortby: undefined,
      page: "1",
      actionType: "initialLoad",
    },
    message: "Invalid error message JSON returned",
    status: -1,
  };
}

export default function SearchError({ error, reset }: ErrorProps) {
  const t = useTranslations("Search");
  const searchParams = useSearchParams();
  const previousSearchParams =
    usePrevious<ReadonlyURLSearchParams>(searchParams);

  useEffect(() => {
    if (
      reset &&
      previousSearchParams &&
      searchParams.toString() !== previousSearchParams?.toString()
    ) {
      reset();
    }
  }, [searchParams, reset, previousSearchParams]);

  useEffect(() => {
    console.error(error);
  }, [error]);

  // The error message is passed as an object that's been stringified.
  // Parse it here.
  let parsedErrorData;

  if (!isValidJSON(error.message)) {
    // the error likely is just a string with a non-specific Server Component error when running the built app
    // "An error occurred in the Server Components render. The specific message is omitted in production builds..."
    parsedErrorData = createBlankParsedError();
  } else {
    // Valid error thrown from server component
    parsedErrorData = JSON.parse(error.message) as ParsedError;
  }
  const convertedSearchParams = convertSearchParamsToProperTypes(
    parsedErrorData.searchInputs,
  );
  const { agency, category, eligibility, fundingInstrument, query, status } =
    convertedSearchParams;

  // note that the validation error will contain untranslated strings
  const ErrorAlert =
    parsedErrorData.details && parsedErrorData.type === "ValidationError" ? (
      <Alert type="error" heading={t("validationError")} headingLevel="h4">
        {`Error in ${parsedErrorData.details.field || "a search field"}: ${parsedErrorData.details.message || "adjust your search and try again"}`}
      </Alert>
    ) : (
      <ServerErrorAlert callToAction={t("generic_error_cta")} />
    );

  return (
    <QueryProvider>
      <div className="grid-container">
        <div className="search-bar">
          <SearchBar query={query} />
        </div>
        <div className="grid-row grid-gap">
          <div className="tablet:grid-col-4">
            <ContentDisplayToggle
              showCallToAction={t("filterDisplayToggle.showFilters")}
              hideCallToAction={t("filterDisplayToggle.hideFilters")}
              breakpoint={Breakpoints.TABLET}
            >
              <SearchFilters
                opportunityStatus={status}
                eligibility={eligibility}
                category={category}
                fundingInstrument={fundingInstrument}
                agency={agency}
              />
            </ContentDisplayToggle>
          </div>
          <div className="tablet:grid-col-8">{ErrorAlert}</div>
        </div>
      </div>
    </QueryProvider>
  );
}
