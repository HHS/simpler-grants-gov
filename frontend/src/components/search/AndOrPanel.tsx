"use client";

import { useSearchParamUpdater } from "src/hooks/useSearchParamUpdater";

import { useCallback } from "react";

import { andOrOptions } from "./SearchFilterAccordion/SearchFilterOptions";
import { SearchFilterRadio } from "./SearchFilterRadio";

export function AndOrPanel({
  hasSearchTerm,
  andOrParam,
}: {
  hasSearchTerm: boolean;
  andOrParam: Set<string>;
}) {
  const { setStaticQueryParam, setQueryParam } = useSearchParamUpdater();

  const toggleAndOrSelection = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      if (hasSearchTerm) {
        return setQueryParam("andOr", event.target.value);
      }
      setStaticQueryParam("andOr", event.target.value);
    },
    [hasSearchTerm],
  );

  return andOrOptions.map((option) => {
    return (
      <SearchFilterRadio
        id={option.id}
        name={option.value}
        label={option.label}
        onChange={toggleAndOrSelection}
        value={option.value}
        checked={
          option.value === "and"
            ? !andOrParam.size || andOrParam.has(option.value)
            : andOrParam.has(option.value)
        }
      />
    );
  });

  // return (
  //   <div>
  //     <SearchFilterRadio
  //       id="andOr-"
  //       name={`Any ${title}`}
  //       label={`Any ${title.toLowerCase()}`}
  //       onChange={toggleAndOrSelection}
  //       value={undefined}
  //       facetCount={undefined}
  //       checked={isNoneSelected}
  //     />
  //     <SearchFilterRadio />
  //   </div>
  // );
}
