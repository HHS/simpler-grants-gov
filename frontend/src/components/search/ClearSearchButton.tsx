"use client";

import { useSearchParamUpdater } from "src/hooks/useSearchParamUpdater";

import { Button } from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

export function ClearSearchButton({
  buttonText,
  includeIcon = false,
  className,
  paramsToClear,
}: {
  buttonText: string;
  includeIcon?: boolean;
  className?: string;
  paramsToClear?: string[];
}) {
  const { clearQueryParams } = useSearchParamUpdater();
  return (
    <Button
      unstyled
      type="button"
      onClick={() => clearQueryParams(paramsToClear)}
      className={className}
    >
      {includeIcon && (
        <USWDSIcon
          name="close"
          className="margin-right-05 margin-left-neg-05"
        />
      )}
      {buttonText}
    </Button>
  );
}
