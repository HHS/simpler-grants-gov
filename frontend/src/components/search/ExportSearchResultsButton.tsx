"use client";

import { downloadSearchResultsCSV } from "src/services/fetch/fetchers/clientSearchResultsDownloadFetcher";

import { useTranslations } from "next-intl";
import { useSearchParams } from "next/navigation";
import { useCallback } from "react";
import { Button } from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

export function ExportSearchResultsButton() {
  const t = useTranslations("Search.exportButton");
  const searchParams = useSearchParams();

  const downloadSearchResults = useCallback(() => {
    // catch included here to satisfy linter
    downloadSearchResultsCSV(searchParams).catch((e) => {
      throw e;
    });
  }, [searchParams]);

  return (
    <div
      className="desktop:grid-col-4 desktop:display-flex flex-align-self-center"
      data-testid="search-download-button-container"
    >
      <Button
        outline={true}
        type={"submit"}
        className="width-auto margin-top-2 tablet:width-100 tablet-lg:margin-top-0"
        onClick={downloadSearchResults}
      >
        <USWDSIcon name="file_download" className="usa-icon--size-3" />
        {t("title")}
      </Button>
    </div>
  );
}
