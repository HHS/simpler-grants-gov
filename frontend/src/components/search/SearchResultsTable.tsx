import clsx from "clsx";
import { isNil } from "lodash";
import { fetchSavedOpportunities } from "src/services/fetch/fetchers/savedOpportunityFetcher";
import {
  BaseOpportunity,
  OpportunityStatus,
} from "src/types/opportunity/opportunityResponseTypes";
import { SearchResponseData } from "src/types/search/searchRequestTypes";
import { toShortMonthDate } from "src/utils/dateUtil";
import { formatCurrency } from "src/utils/formatCurrencyUtil";
import { getOpportunityUrl } from "src/utils/opportunity/opportunityUtils";

import { useTranslations } from "next-intl";

import {
  TableCellData,
  TableWithResponsiveHeader,
} from "src/components/TableWithResponsiveHeader";
import { USWDSIcon } from "src/components/USWDSIcon";

const statusColorClasses = {
  posted: "bg-accent-warm-light",
  forecasted: "bg-accent-warm-lightest",
  closed: "bg-base-lightest",
  archived: "bg-base-lightest",
};

const SearchTableStatusDisplay = ({
  status,
}: {
  status: OpportunityStatus;
}) => {
  const t = useTranslations("Search.table");
  const backgroundClass = statusColorClasses[status];
  return (
    <div
      className={clsx("text-center padding-y-05 minw-15 radius-md", {
        [backgroundClass]: !!backgroundClass,
      })}
    >
      {t(`statuses.${status}`)}
    </div>
  );
};

const CloseDateDisplay = ({ closeDate }: { closeDate: string }) => {
  const t = useTranslations("Search.table");
  return <>{closeDate ? toShortMonthDate(closeDate) : t("tbd")}</>;
};

const TitleDisplay = ({
  opportunity,
  saved,
}: {
  opportunity: BaseOpportunity;
  saved: boolean;
}) => {
  const t = useTranslations("Search.table");
  return (
    <>
      <div className="font-sans-lg text-bold">
        <a href={getOpportunityUrl(opportunity.opportunity_id)}>
          {opportunity.opportunity_title}
        </a>
      </div>
      <div className="display-none tablet-lg:display-block font-sans-xs">
        <span className="text-bold">{t("number")}:</span>{" "}
        {opportunity.opportunity_number}
      </div>
      {saved && (
        <div className="margin-top-2 display-inline-block">
          <span className="display-flex flex-align-center font-sans-2xs">
            <USWDSIcon
              name="star"
              className="text-accent-warm-dark button-icon-md padding-right-05"
            />
            {t("saved")}
          </span>
        </div>
      )}
    </>
  );
};

const AgencyDisplay = ({ opportunity }: { opportunity: BaseOpportunity }) => {
  const t = useTranslations("Search.table");
  return (
    <>
      <div className="tablet-lg:margin-bottom-1">{opportunity.agency_name}</div>
      <div className="font-sans-xs display-none tablet-lg:display-block">
        <span className="text-bold">{t("published")}</span>:{" "}
        {toShortMonthDate(opportunity.summary.post_date || "")}
      </div>
      <div className="font-sans-xs display-none tablet-lg:display-block">
        {t("expectedAwards")}:{" "}
        {isNil(opportunity.summary.expected_number_of_awards)
          ? "--"
          : opportunity.summary.expected_number_of_awards}
      </div>
    </>
  );
};

const toSearchResultsTableRow = (
  result: BaseOpportunity,
  saved: boolean,
): TableCellData[] => {
  return [
    {
      cellData: (
        <CloseDateDisplay closeDate={result.summary.close_date || ""} />
      ),
      stackOrder: 2,
    },
    {
      cellData: <SearchTableStatusDisplay status={result.opportunity_status} />,
      stackOrder: 1,
    },
    {
      cellData: <TitleDisplay opportunity={result} saved={saved} />,
      stackOrder: 0,
    },
    {
      cellData: <AgencyDisplay opportunity={result} />,
      stackOrder: 3,
    },
    {
      cellData: isNil(result.summary.award_floor)
        ? "$--"
        : formatCurrency(result.summary.award_floor),
      stackOrder: 4,
    },
    {
      cellData: isNil(result.summary.award_ceiling)
        ? "$--"
        : formatCurrency(result.summary.award_ceiling),
      stackOrder: 5,
    },
  ];
};

export const SearchResultsTable = async ({
  searchResults,
}: {
  searchResults: SearchResponseData;
}) => {
  const t = useTranslations("Search.table");

  const savedOpportunities = await fetchSavedOpportunities();
  const savedOpportunityIds = savedOpportunities.map(
    (opportunity) => opportunity.opportunity_id,
  );

  const headerContent: TableCellData[] = [
    { cellData: t("headings.closeDate") },
    { cellData: t("headings.status") },
    { cellData: t("headings.title") },
    { cellData: t("headings.agency") },
    { cellData: t("headings.awardMin") },
    { cellData: t("headings.awardMax") },
  ];
  const tableRowData = searchResults.map((result) =>
    toSearchResultsTableRow(
      result,
      savedOpportunityIds.includes(result.opportunity_id),
    ),
  );
  return (
    <TableWithResponsiveHeader
      headerContent={headerContent}
      tableRowData={tableRowData}
    />
  );
};
