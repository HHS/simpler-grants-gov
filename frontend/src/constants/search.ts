import { FrontendFilterNames } from "src/types/search/searchFilterTypes";

// Show all values status for search.
export const SEARCH_NO_STATUS_VALUE = "none";
export const SEARCH_DEFAULT_VALUES = ["forecasted", "posted"];
export const defaultFilterValues: { [key in FrontendFilterNames]?: string[] } =
  {
    status: SEARCH_DEFAULT_VALUES,
  };
