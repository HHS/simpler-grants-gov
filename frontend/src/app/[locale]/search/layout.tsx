import { SEARCH_CRUMBS } from "src/constants/breadcrumbs";
import { LayoutProps } from "src/types/generalTypes";

import { useTranslations } from "next-intl";
import { setRequestLocale } from "next-intl/server";
import { use } from "react";

import BetaAlert from "src/components/BetaAlert";
import Breadcrumbs from "src/components/Breadcrumbs";
import SearchCallToAction from "src/components/search/SearchCallToAction";

export default function SearchLayout({ children, params }: LayoutProps) {
  const { locale } = use(params);
  setRequestLocale(locale);

  const t = useTranslations("Search.beta_alert");

  return (
    <>
      <BetaAlert
        containerClasses="margin-top-5"
        heading={t("alert_title")}
        alertMessage={t.rich("alert", {
          mailToGrants: (chunks) => (
            <a href="mailto:simpler@grants.gov">{chunks}</a>
          ),
          bugReport: (chunks) => (
            <a href="https://github.com/HHS/simpler-grants-gov/issues/new?template=1_bug_report.yml">
              {chunks}
            </a>
          ),
          featureRequest: (chunks) => (
            <a href="https://github.com/HHS/simpler-grants-gov/issues/new?template=2_feature_request.yml">
              {chunks}
            </a>
          ),
        })}
      />
      <Breadcrumbs breadcrumbList={SEARCH_CRUMBS} />
      <SearchCallToAction />
      {children}
    </>
  );
}
