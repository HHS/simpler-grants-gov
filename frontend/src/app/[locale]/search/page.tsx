import { Metadata } from "next";
import QueryProvider from "src/app/[locale]/search/QueryProvider";
import { environment } from "src/constants/environments";
import withFeatureFlag from "src/hoc/withFeatureFlag";
import { LocalizedPageProps } from "src/types/intl";
import { SearchParamsTypes } from "src/types/search/searchRequestTypes";
import { Breakpoints } from "src/types/uiTypes";
import { convertSearchParamsToProperTypes } from "src/utils/search/convertSearchParamsToProperTypes";

import { useTranslations } from "next-intl";
import { getTranslations, setRequestLocale } from "next-intl/server";
import { redirect } from "next/navigation";
import { use } from "react";

import ContentDisplayToggle from "src/components/ContentDisplayToggle";
import { SaveSearchPanel } from "src/components/search/SaveSearchPanel";
import SearchAnalytics from "src/components/search/SearchAnalytics";
import SearchBar from "src/components/search/SearchBar";
import SearchFilters from "src/components/search/SearchFilters";
import SearchResults from "src/components/search/SearchResults";

export async function generateMetadata({ params }: LocalizedPageProps) {
  const { locale } = await params;
  const t = await getTranslations({ locale });
  const meta: Metadata = {
    title: t("Search.title"),
    description: t("Search.meta_description"),
  };
  return meta;
}
type SearchPageProps = {
  searchParams: Promise<SearchParamsTypes>;
  params: Promise<{ locale: string }>;
};

function Search({ searchParams, params }: SearchPageProps) {
  const { locale } = use(params);
  const resolvedSearchParams = use(searchParams);
  setRequestLocale(locale);
  const t = useTranslations("Search");

  const convertedSearchParams =
    convertSearchParamsToProperTypes(resolvedSearchParams);
  const { agency, category, eligibility, fundingInstrument, query, status } =
    convertedSearchParams;

  if (!("page" in resolvedSearchParams)) {
    resolvedSearchParams.page = "1";
  }

  return (
    <>
      <SearchAnalytics
        params={resolvedSearchParams}
        newRelicEnabled={environment.NEW_RELIC_ENABLED === "true"}
      />
      <QueryProvider>
        <div className="grid-container">
          <div className="search-bar">
            <SearchBar query={query} />
          </div>
          <div className="grid-row grid-gap">
            <div className="tablet:grid-col-4">
              <ContentDisplayToggle
                showCallToAction={t("filterDisplayToggle.showFilters")}
                hideCallToAction={t("filterDisplayToggle.hideFilters")}
                breakpoint={Breakpoints.TABLET}
                type="centered"
              >
                <SaveSearchPanel />
                <SearchFilters
                  opportunityStatus={status}
                  eligibility={eligibility}
                  category={category}
                  fundingInstrument={fundingInstrument}
                  agency={agency}
                />
              </ContentDisplayToggle>
            </div>
            <div className="tablet:grid-col-8">
              <SearchResults
                searchParams={convertedSearchParams}
                query={query}
                loadingMessage={t("loading")}
              ></SearchResults>
            </div>
          </div>
        </div>
      </QueryProvider>
    </>
  );
}

// Exports page behind both feature flags
export default withFeatureFlag<SearchPageProps, never>(
  Search,
  "searchOff",
  () => redirect("/maintenance"),
);
