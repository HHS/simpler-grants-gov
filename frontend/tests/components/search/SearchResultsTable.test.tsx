import { render, screen, within } from "@testing-library/react";
import { axe } from "jest-axe";
import { mockOpportunity } from "src/utils/testing/fixtures";
import { useTranslationsMock } from "src/utils/testing/intlMocks";

import { SearchResultsTable } from "src/components/search/SearchResultsTable";

const mockFetchSavedOpportunities = jest.fn();

jest.mock("next-intl", () => ({
  useTranslations: () => useTranslationsMock(),
}));

jest.mock("src/services/fetch/fetchers/savedOpportunityFetcher", () => ({
  fetchSavedOpportunities: () => mockFetchSavedOpportunities() as unknown,
}));

// this does not directly test responsive aspects of the component, that should be done in e2e tests
// see https://github.com/HHS/simpler-grants-gov/issues/5414
describe("SearchResultsTable", () => {
  beforeEach(() =>
    mockFetchSavedOpportunities.mockResolvedValue([{ opportunity_id: 1 }]),
  );
  afterEach(() => jest.resetAllMocks());
  it("passes accessibility test", async () => {
    const component = await SearchResultsTable({
      searchResults: [mockOpportunity],
    });
    const { container } = render(component);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
  // skip until we're ready
  it.skip("matches snapshot", async () => {
    const component = await SearchResultsTable({
      searchResults: [mockOpportunity],
    });
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
  it("displays saved search indicator when applicable", async () => {
    const component = await SearchResultsTable({
      searchResults: [
        mockOpportunity,
        { ...mockOpportunity, opportunity_id: 1 },
      ],
    });
    render(component);

    const results = screen.getAllByRole("row");
    expect(results).toHaveLength(3); // first is the header row
    expect(within(results[1]).queryByText("saved")).not.toBeInTheDocument();
    expect(within(results[2]).getByText("saved")).toBeInTheDocument();
  });
  it("displays headings for all columns as expected", async () => {
    const component = await SearchResultsTable({
      searchResults: [mockOpportunity],
    });
    render(component);

    expect(
      screen.getByRole("columnheader", { name: "headings.closeDate" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("columnheader", { name: "headings.status" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("columnheader", { name: "headings.title" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("columnheader", { name: "headings.agency" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("columnheader", { name: "headings.awardMin" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("columnheader", { name: "headings.awardMax" }),
    ).toBeInTheDocument();
  });
  // note that values for this test also include the header value - this is a remnant of the responsive experience,
  // Jest doesn't know that these header values should be hidden :shrug:
  it("displays data for all columns as expected", async () => {
    const component = await SearchResultsTable({
      searchResults: [
        {
          ...mockOpportunity,
          summary: { ...mockOpportunity.summary, expected_number_of_awards: 1 },
        },
      ],
    });
    render(component);

    const results = screen.getAllByRole("row");
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.closeDate Feb 1, 2023",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.status statuses.posted",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.title Test Opportunity number: OPP-12345",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.agency published : Jan 15, 2023 expectedAwards: 1",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.awardMin $10,000",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.awardMax $50,000",
      }),
    ).toBeInTheDocument();
  });

  it("handles display of empty values", async () => {
    const component = await SearchResultsTable({
      searchResults: [
        {
          ...mockOpportunity,
          summary: {
            ...mockOpportunity.summary,
            close_date: "",
            award_ceiling: null,
            award_floor: null,
          },
        },
      ],
    });
    render(component);

    const results = screen.getAllByRole("row");
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.closeDate tbd",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.agency published : Jan 15, 2023 expectedAwards: --",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.awardMin $--",
      }),
    ).toBeInTheDocument();
    expect(
      within(results[1]).getByRole("cell", {
        name: "headings.awardMax $--",
      }),
    ).toBeInTheDocument();
  });
});
