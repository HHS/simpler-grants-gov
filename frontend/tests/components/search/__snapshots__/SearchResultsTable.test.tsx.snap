// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsTable matches snapshot 1`] = `
<div>
  <table
    class="usa-table usa-table--borderless simpler-responsive-table width-full tablet-lg:width-auto border-base tablet-lg:border-0"
    data-testid="table"
  >
    <thead>
      <tr>
        <th
          class="bg-base-lightest padding-y-205 minw-15"
          scope="col"
        >
          headings.closeDate
        </th>
        <th
          class="bg-base-lightest padding-y-205 minw-15"
          scope="col"
        >
          headings.status
        </th>
        <th
          class="bg-base-lightest padding-y-205 minw-15"
          scope="col"
        >
          headings.title
        </th>
        <th
          class="bg-base-lightest padding-y-205 minw-15"
          scope="col"
        >
          headings.agency
        </th>
        <th
          class="bg-base-lightest padding-y-205 minw-15"
          scope="col"
        >
          headings.awardMin
        </th>
        <th
          class="bg-base-lightest padding-y-205 minw-15"
          scope="col"
        >
          headings.awardMax
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="border-base border-x border-y tablet-lg:border-0"
      >
        <td
          class="tablet-lg:display-table-cell border-base order-2 tablet-lg:order-0 display-block"
        >
          <div
            class="display-flex flex-align-center"
          >
            <div
              class="tablet-lg:display-none flex-1 text-bold"
              data-testid="responsive-header-0-0"
            >
              headings.closeDate
            </div>
            <div
              class="flex-2"
              data-testid="responsive-data-0-0"
            >
              Feb 1, 2023
            </div>
          </div>
        </td>
        <td
          class="tablet-lg:display-table-cell border-base order-1 tablet-lg:order-1 display-block"
        >
          <div
            class="display-flex flex-align-center"
          >
            <div
              class="tablet-lg:display-none flex-1 text-bold"
              data-testid="responsive-header-0-1"
            >
              headings.status
            </div>
            <div
              class="flex-2"
              data-testid="responsive-data-0-1"
            >
              <div
                class="text-center padding-y-05 minw-15 radius-md bg-accent-warm-light"
              >
                statuses.posted
              </div>
            </div>
          </div>
        </td>
        <td
          class="tablet-lg:display-table-cell border-base order-0 tablet-lg:order-2 display-block"
        >
          <div
            class="display-flex flex-align-center"
          >
            <div
              class="tablet-lg:display-none flex-1 text-bold"
              data-testid="responsive-header-0-2"
            >
              headings.title
            </div>
            <div
              class="flex-2"
              data-testid="responsive-data-0-2"
            >
              <div
                class="font-sans-lg text-bold"
              >
                <a
                  href="/opportunity/63588df8-f2d1-44ed-a201-5804abba696a"
                  id="search-result-link-1-1"
                >
                  Test Opportunity
                </a>
              </div>
              <div
                class="display-none tablet-lg:display-block font-sans-xs"
              >
                <span
                  class="text-bold"
                >
                  number
                  :
                </span>
                 
                OPP-12345
              </div>
            </div>
          </div>
        </td>
        <td
          class="tablet-lg:display-table-cell border-base order-3 tablet-lg:order-3 display-block"
        >
          <div
            class="display-flex flex-align-center"
          >
            <div
              class="tablet-lg:display-none flex-1 text-bold"
              data-testid="responsive-header-0-3"
            >
              headings.agency
            </div>
            <div
              class="flex-2"
              data-testid="responsive-data-0-3"
            >
              <div
                class="tablet-lg:margin-bottom-1"
              />
              <div
                class="font-sans-xs display-none tablet-lg:display-block"
              >
                <span
                  class="text-bold"
                >
                  published
                </span>
                :
                 
                Jan 15, 2023
              </div>
              <div
                class="font-sans-xs display-none tablet-lg:display-block"
              >
                expectedAwards
                :
                 
                --
              </div>
            </div>
          </div>
        </td>
        <td
          class="tablet-lg:display-table-cell border-base order-4 tablet-lg:order-4 display-block"
        >
          <div
            class="display-flex flex-align-center"
          >
            <div
              class="tablet-lg:display-none flex-1 text-bold"
              data-testid="responsive-header-0-4"
            >
              headings.awardMin
            </div>
            <div
              class="flex-2"
              data-testid="responsive-data-0-4"
            >
              $10,000
            </div>
          </div>
        </td>
        <td
          class="tablet-lg:display-table-cell border-base order-5 tablet-lg:order-5 display-block"
        >
          <div
            class="display-flex flex-align-center"
          >
            <div
              class="tablet-lg:display-none flex-1 text-bold"
              data-testid="responsive-header-0-5"
            >
              headings.awardMax
            </div>
            <div
              class="flex-2"
              data-testid="responsive-data-0-5"
            >
              $50,000
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
